### PR TITLE
[SKB-2408] Fixed show_in_fs function for Windows when using drive letters.

### PR DESCRIPTION
**Pull request by rob-aitchison**

### CODE REVIEW

**Review by staceyoue — APPROVED** [review]
👍

**Comment by rob-aitchison** [comment]
Hello @mathurf , I just wanted to get your approval on this one as well. I parked it until the new year but this change was tested on all three platforms at the end of last year. Let me know if anything is unclear or if you have any other questions?
Thanks!

File: python/tk_multi_about/dialog.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def show_in_fs(self):
         paths = self._app.context.filesystem_locations
         for disk_location in paths:
             if not QtGui.QDesktopServices.openUrl(
-                "file://{0}".format(disk_location).replace("\\", "/")
+                "file:///{0}".format(disk_location).replace("\\", "/")
```
Comment (pscadding) [inline python/tk_multi_about/dialog.py:107]: I'm not an expert on this file URI format, but according to here:
https://en.wikipedia.org/wiki/File_URI_scheme
The following are valid:

> file:/path, file:///path or file://hostname/path`

What concerns me a bit about this change is, what if the disk_location starts with a `/`?
You then end up with four does that still work?
What is the issue this is trying to fix?

Comment (mathurf) [reply python/tk_multi_about/dialog.py:107]: I just tried with my advanced setup on Windows who start with \\[machine name]\...\
and it fails with this error: // Error: Shotgun tk-multi-about: Failed to open '\\remmj06fbt9\G\ShotgunNiruhtam\mnt\projects\testcreate'! //

**Review by mathurf — CHANGES_REQUESTED** [review]
I just tried with my advanced setup on Windows who start with `\\[machine name]\...\`
and it fails with this error: `// Error: Shotgun tk-multi-about: Failed to open '\\remmj06fbt9\G\ShotgunNiruhtam\mnt\projects\testcreate'! //`

File: python/tk_multi_about/dialog.py (hunk 1 above)
Comment (rob-aitchison) [reply python/tk_multi_about/dialog.py:107]: @pscadding It is currently broken on Windows with letter drive mapping.
@mathurf thanks for that, I will test some more cases with UNC, etc.

**Review by rob-aitchison — COMMENTED** [review]
Ready to re-test now, thanks!

**Review by mathurf — APPROVED** [review]
On my side I confirm that now it is working with both, UNC path and with a drive letter.
If Phil is ok with the change then :shipit:

File: python/tk_multi_about/dialog.py [hunk 2]
```diff
@@ -106,7 +106,16 @@ def show_in_fs(self):
...
-                self._app.log_error("Failed to open '%s'!" % disk_location)
+                # You will have a False return here on Windows when using
+                # drive letters and not UNC paths - add third slash
+                if not QtGui.QDesktopServices.openUrl(
+                    "file:///{0}".format(disk_location).replace("\\", "/")
+                ):
+                    self._app.log_error(
+                        "Failed to open system file browser for '%s'!" % disk_location
+                    )
+                    break
```
Comment (pscadding) [inline python/tk_multi_about/dialog.py:117]: Adding a break in here changes the behaviour I think. The comment on line 103 says we are launching a window for each location, but adding a break would mean we would stop trying to open locations if we found one locations that  didn't open.
I think it seems reasonable to try and open other locations even if one of them failed, so maybe we shouldn't have a break here, what do you think?

**Review by pscadding — CHANGES_REQUESTED** [review]
I have a small change to suggest, but other wise looks great!

File: python/tk_multi_about/dialog.py (hunk 2 above)
Comment (rob-aitchison) [reply python/tk_multi_about/dialog.py:117]: I totally agree @pscadding and will make that change. What do you think of the added log_info line right below it? Do you think we should keep it as info or set it up as a debug log only? Thanks!

Comment (pscadding) [reply python/tk_multi_about/dialog.py:117]: Yeah I considered that line as well, but tbh I'm indifferent, its a low volume operation so its not likely to clog up logs and isn't going to impact on performance. 
If you think there is value in being always logged then I'm fine with it being info.

**Review by pscadding — APPROVED** [review]
Looks great!
If I'm being picky though `log_debug` is deprecated in favour of `logger.debug`
https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=log_debug#sgtk.platform.Application.log_debug
Sorry should have mentioned before 😞

**Comment by rob-aitchison** [comment]
@pscadding I will re-factor the new logging now, no worries, cheers!

**Pull request merged**